src/dlite-type.c

```diff
@@ -718,18 +718,18 @@ int dlite_type_print(char *dest, size_t n, const void *p, DLiteType dtype,
   case dliteBlob:
     if (!(qflags & strquoteNoQuote)) {
         int v = snprintf(dest+m, PDIFF(n, m), "\"");
-        if (v < 0) return err(-1, "error printing initial quote for blob");
+        if (v < 0) return err(dliteIOError, "error printing initial quote for blob");
```

Comment:
Hmm, I don't think this should be a `dliteIOError`, since `snprintf()` prints to memory and not to a file or stream. 

This error is not supposed to happen, so it could be a `dliteRuntimeError`, but a dedicated `dliteFormatError` could be another possibility.

Reply:
dliteFormatError added